### PR TITLE
docs(devkit): add guides on devkit

### PR DESCRIPTION
**Pull request by vsavkin**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: docs/angular/guides/nx-devkit-angular-devkit.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Nx Devkit and Angular Devkit
+
+Nx comes with a devkit to write generators and executors, but you can also use Angular devkit (schematics and builders). An Angular schematic is a second way to implement generators. An Angular builder is a second way to implement an executor.
+
+What is the different between Nx Devkit and Angular Devkit?
```
Comment (brandonroberts) [inline docs/angular/guides/nx-devkit-angular-devkit.md:5]:
```suggestion
What is the difference between Nx Devkit and Angular Devkit?
```

File: docs/angular/guides/nx-devkit-angular-devkit.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+```
+
+**Some notable changes:**
+
+- Nx Devkit generators do not use partial application. An Angular Schematic returns a rule that is then invoked with a host.
+- Nx Devkit generators do not use `chain`. Just invoke the helpers directly. This makes them more debuggable. As you step through the generator you can see the host being updated.
+- `chain([mergeWith(apply(url` is replaced with `generateFiles` :)
+- Nx Devkit generators return a function that performs side effects. Angular Schematics have to create a custom task runner and register a task using it.
+- You don't need any special helpers ot compose Nx Devkit generators. You do need to go through special resolution step to compose schematics.
```
Comment (brandonroberts) [inline docs/angular/guides/nx-devkit-angular-devkit.md:76]:
```suggestion
- You don't need any special helpers to compose Nx Devkit generators. You do need to go through a special resolution step to compose schematics.
```

File: docs/angular/guides/nx-devkit-angular-devkit.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  if (options.allCaps) {
+    console.log(options.message.toUpperCase());
+  } else {
+    console.log(options.message);
+  }
+  return { success: true };
+}
+```
+
+Same executor written an Angular builder:
```
Comment (brandonroberts) [inline docs/angular/guides/nx-devkit-angular-devkit.md:104]:
```suggestion
Same executor written as an Angular builder:
```

File: docs/angular/guides/nx-devkit-angular-devkit.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    console.log(options.message);
+  }
+  return of({ success: true });
+}
+export default createBuilder<NextBuildBuilderOptions>(run);
+```
+
+Some notable changes:
+
+- Nx Devkit executors return promises. If you want, you can always covert an observable to a promise.
```
Comment (brandonroberts) [inline docs/angular/guides/nx-devkit-angular-devkit.md:128]:
```suggestion
- Nx Devkit executors return promises. If you want, you can always convert an observable to a promise.
```

File: docs/shared/devkit.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+  }
+  return { success: true };
+}
+```
+
+The context parameters contains information about the workspace and the invoked target.
+
+## Learn More
+
+- Learn more about using workspace generators in [this guide](/{{framework}}/workspace/generators/workspace-generators)
```
Comment (brandonroberts) [inline docs/shared/devkit.md:203]:
```suggestion
- Learn more about using workspace generators in [this guide](/{{framework}}/latest/workspace/generators/workspace-generators)
```

File: docs/shared/devkit.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+- Learn more about writing custom plugins in [this guide](/{{framework}}/plugins/nx-plugin/overview)
```
Comment (brandonroberts) [inline docs/shared/devkit.md:204]:
```suggestion
- Learn more about writing custom plugins in [this guide](/{{framework}}/latest/plugins/nx-plugin/overview)
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by vsavkin** [comment]
Fixed it. Thank you, @brandonroberts

**Pull request merged**